sklearn/pipeline.py

```diff
@@ -2046,6 +2046,14 @@ def _hstack(self, Xs):
 
         adapter = _get_container_adapter("transform", self)
         if adapter and all(adapter.is_supported_container(X) for X in Xs):
+            # Rename all columns to avoid duplicated column names.
+            # The names are not important here as final column names will be
+            # generated by the set_output wrapper using `get_feature_names_out`.
+            Xs = (
+                adapter.rename_columns(X, [f"{i}{j}" for j in range(X.shape[1])])
```

Comment:
```suggestion
                adapter.rename_columns(X, [f"{i}_{j}" for j in range(X.shape[1])])
```

otherwise maybe we can get a duplicate for column 11 of dataframe 1 and column 1 of dataframe 11?

Reply:
good catch, thanks :)